packages/services/hmi-server/docker/native/Dockerfile.native

```diff
@@ -13,15 +9,30 @@
 #
 # docker run -i --rm -p 8080:8080 quarkus/server-scaffold
 #
-###
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6-994
-WORKDIR /work/
-RUN chown 1001 /work \
-    && chmod "g+rwX" /work \
-    && chown 1001:root /work
-COPY --chown=1001:root build/*-runner /work/application
 
-EXPOSE 8080
-USER 1001
+## Stage 1 : build with gradle builder image with native capabilities
+FROM quay.io/quarkus/ubi-quarkus-graalvmce-builder-image:22.3-java17 AS builder
+USER root
+RUN microdnf install findutils
+
+# Copy over necessary gradle pieces
+COPY --chown=quarkus:quarkus gradlew /code/gradlew
+COPY --chown=quarkus:quarkus gradle /code/gradle
+COPY --chown=quarkus:quarkus settings.gradle /code/
+COPY --chown=quarkus:quarkus gradle.properties /code/
+COPY --chown=quarkus:quarkus packages/services/hmi-server/build.gradle /code/
 
+# Copy over the source files for building
+USER quarkus
+WORKDIR /code
+COPY packages/services/hmi-server/src /code/src
+
+RUN ./gradlew build -Dquarkus.package.type=native
+
+## Stage 2 : create the docker final image
+FROM quay.io/quarkus/quarkus-micro-image:2.0
+WORKDIR /work/
+COPY --from=builder /code/build/*-runner /work/application
+RUN chmod 775 /work
+EXPOSE 8080
 CMD ["./application", "-Dquarkus.http.host=0.0.0.0"]
```

Comment:
Is this because Quarkus by default runs on port `8080`? then we port the all income connection from `localhost` to it?

Reply:
@dgauldie ?

Reply:
That should be an error, a hold over from "normal" Java deployments.  That `EXPOSE` should specify `3000` not 8080